### PR TITLE
Model test

### PR DESCRIPTION
**Pull request by kazim110**
1. Add model test for user
2. Add model test for space
3. Add model test for reservation
4. Add model test for city
5. Add model test for state
6. Add model test for country

Screenshot 2024-03-21 001111

### CODE REVIEW

File: spec/models/reservation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      expect(reservation.errors[:start_date]).to include("can't be blank")
+    end
+
+    it "should validate presence of end_date" do
+      reservation = Reservation.new(date_reserved: Date.today, start_date: Date.today)
+      expect(reservation).not_to be_valid
+      expect(reservation.errors[:end_date]).to include("can't be blank")
+    end
+  end
+end
```
Comment (Alejandroq12) [inline spec/models/reservation_spec.rb:40]: Good job, testing the reservation model.
Correctly testing validations, and associations.

File: spec/models/space_cw_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      space_cw = SpaceCw.new(name: "Test Name", description: "Test Description", discount: "not_a_number")
+      expect(space_cw).not_to be_valid
+      expect(space_cw.errors[:discount]).to include("is not a number")
+
+      space_cw.discount = -1
+      expect(space_cw).not_to be_valid
+      expect(space_cw.errors[:discount]).to include("must be greater than or equal to 0")
+    end
+  end
+end
```
Comment (Alejandroq12) [inline spec/models/space_cw_spec.rb:49]: Good job, testing the SpaceCw model.
Correctly testing validations, and associations.

**Review by Alejandroq12 — APPROVED** [review]
Good job. Changes are approved.
You are correctly testing models which is best practice to minimize the risk of unexpected bugs

It is time to merge it.

**Comment by Alejandroq12** [comment]
Hello, @kazim110. Please, fix linter errors before merging.

**Review by jcmunav63 — APPROVED** [review]
Hello, @kazim110 :

You did a very good job implementing the unit tests for the models on our Ruby on Rails back-end.

You have implemented well all the expected associations and validations for these models: User, Space_Cw, Reservation, Country, State, City.

You did a great job. There's nothing else to say but... It's time to merge it! 👍🏽 💯 ✅

**Pull request merged**